### PR TITLE
[SNOW-643653] Fix Citi Blackduck vulnerabilities 

### PR DESCRIPTION
**Pull request by sfc-gh-rcheng**
Updated the following packages:
apache kafka connect api: 3.2.0 -> 3.2.3
apache kafka clients: 2.8.1 -> 2.8.2
apache avro: 1.11.0 -> 1.11.1
apache avro maven plugin: 1.8.1 -> 1.11.1
confluent avro serializer: 5.5.1 -> 7.2.1

### CODE REVIEW

File: test/perf_test/pom.xml [hunk 1]
```diff
@@ -25,7 +25,7 @@
     <dependency>
       <groupId>io.confluent</groupId>
       <artifactId>kafka-avro-serializer</artifactId>
-      <version>5.1.2</version>
```
Comment (sfc-gh-japatel) [inline test/perf_test/pom.xml:28]: FYI. i dont think this will impact CVEs since it is in test folder.  but thanks for the update here.

Comment (sfc-gh-rcheng) [reply test/perf_test/pom.xml:28]: i agree, but good to keep test up to date with the actual pom.xml. thanks!

File: pom.xml [hunk 2]
```diff
@@ -290,7 +290,7 @@
         <dependency>
             <groupId>org.apache.kafka</groupId>
             <artifactId>connect-api</artifactId>
-            <version>3.2.0</version>
```
Comment (sfc-gh-japatel) [inline pom.xml:293]: Lets make a change in our documentation. we still mention we use 2.8.1 (unfortunately)
CC: @sfc-gh-lema https://docs.snowflake.com/en/user-guide/kafka-connector-install.html#installation-prerequisites (Search for 2.8.1)

We should also fix what all AK and confluent versions we support for brokers. we have tests against 5.5.11, 6.2.x and 7.2.x. Our documentation is very old and outdated.

**Review by sfc-gh-japatel — APPROVED** [review]
lgtm! thanks

**Review by sfc-gh-tzhang — APPROVED** [review]

File: pom.xml (hunk 2 above)
Comment (sfc-gh-lema) [reply pom.xml:293]: @sfc-gh-japatel thanks for letting me know. I'll create a JIRA and update the documentation.

**Pull request merged**